astoria/common/metadata.py

```diff
@@ -32,15 +34,37 @@ def init(cls, config: AstoriaConfig) -> 'Metadata':
         Based on information from software and the astoria configuration.
         """
         uname = platform.uname()
+        os_release = cls.get_os_version_info()
         return cls(
             astoria_version=__version__,
             kernel_version=uname.release,
             arch=uname.machine,
             python_version=platform.python_version(),
             libc_ver="".join(platform.libc_ver()),
             usercode_entrypoint=config.astprocd.default_usercode_entrypoint,
+            os_name=os_release['NAME'],
+            os_pretty_name=os_release['PRETTY_NAME'],
+            os_version=os_release['VERSION_ID'],
         )
 
+    @classmethod
+    def get_os_version_info(cls) -> Dict[str, str]:
```

Comment:
It would be great to add some tests for this function.

We could pass in the path to os-release as a default parameter so that we can point the suite at some test data?